system/mp/mpbase_mpi.cpp

```diff
@@ -0,0 +1,266 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2012 HPCC Systems®.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+#define mp_decl DECL_EXPORT
+#include "platform.h"
+#include "jlib.hpp"
+#include "jlog.hpp"
+
+#include "mpbase.hpp"
+#include "mpi_wrapper.hpp"
+
+static INode *MyNode=NULL;
+static INode *NullNode=NULL;
+
+
+IGroup *createIGroup(rank_t num,INode **nodes){
```

Comment:
I'm guessing you mean putting spaces instead of tabs right? I will replace tab with 4 spaces. I switched to eclipse editor about a week back and forgot to update coding style in the editor preference. so there's probably some tab spaces here and there.

Reply:
yes 4 space indentation and open/close braces on newlines.

Reply:
But the open brace in a new line is soooo ugly :'(. But will fix it.